### ISSUE TITLE
puma.state doesn't exists masquerading Errno::ENOENT error

### ISSUE THREAD

**Issue by vizcay** [ylecuyer/puma-status#5]
While trying to debug a puma server problem found your tool but I wasn't unable to use it in my server:

```
deploy@xxx:~/applications/xxx/shared/tmp/pids$ puma-status puma.state

puma.state doesn't exists
deploy@xxx:~/applications/xxx/shared/tmp/pids$ cat puma.state
---
pid: 1727
control_url: unix://tmp/sockets/pumactl.sock
control_auth_token: '12345'
deploy@xxx:~/applications/xxx/shared/tmp/pids$ ls -l puma.state
-rw-r--r-- 1 deploy sudo 87 May 18 01:52 puma.state
deploy@xxx:~/applications/xxx/shared/tmp/pids$
```

But I'm sure the file exists and is readable so I think that https://github.com/ylecuyer/puma-status/blob/cd0a4168afc7e621be12e1422291cc6806e9a7a7/lib/puma-status.rb#L20 is getting the `Errno::ENOENT` from another source.

Any ideas? Thanks

**Comment by ylecuyer** [comment]
I managed to reproduce locally, it looks like puma-status doesn't handle well relative paths for the unix socket file and the rescue is too broad and hides this failure as if puma.state was missing.

**Comment by ylecuyer** [comment]
This one will be difficult to handle, puma-status is made to be run from everywhere on the system and here we have a relative path without the root directory where we should start looking from.

In your case you will have to run puma-status from `~/applications/xxx/shared/` to have it working

I'll try to add a warning to make things clear

**Comment by jarthod** [comment]
@ylecuyer why shouldn't it work with relative path like this?

**Comment by ylecuyer** [comment]
It should work but in the puma state file there is no info about the root of relative paths.

For example the following config gives:

```
state_path '/tmp/puma.state'
activate_control_app 'unix://tmp/puma.sock'
```

```
---
pid: 26442
control_url: unix://tmp/puma.sock
control_auth_token: 259a35c85c9764dd84a416b3b96cd84a
```

We can deduce "/home/ylecuyer/Projects/test-puma/tmp/puma.sock" from this

It would be nice to have a `running_from: "/home/ylecuyer/Projects/test-puma/" in the state file

**Comment by jarthod** [comment]
Ah ok so `unix://tmp/puma.sock` is a relative path? looked like absolute to me ;)

**Comment by ylecuyer** [comment]
yes unix socket absolute paths have 3 / : `unix:///tmp/puma.sock` nice for confusion :)

> Anyway if it's relative you can still try from current dir and display an error if it doesn't work?

Yes this is what I did in #7

**Comment by ylecuyer** [comment]
I have published puma-status 1.1 (https://rubygems.org/gems/puma-status/versions/1.1) which should do the trick for your problem until we get https://github.com/puma/puma/pull/2283 sorted out.

I'm closing for now

**Comment by vizcay** [comment]
Great @ylecuyer now I understand why it wasn't working: I've always runned puma-state inside the tmp directory. Tested it now and works like a charm.
